common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -504,6 +529,52 @@ public static void setMemory(long address, long bytes, byte value) {
         PlatformDependent0.setMemory(address, bytes, value);
     }
 
+    public static ByteBuffer allocateDirectNoCleaner(int capacity) {
```

Comment:
add javadoc which indicates `freeDirectNoCleaner` must be used to free this buffer.